component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -104,6 +107,38 @@ public void init() throws IdentityOAuth2Exception {
         }
     }
 
+    /**
+     * get resident Identity Provider
+     *
+     * @param tenantDomain tenant Domain
+     * @param jwtIssuer    issuer extracted from assertion
+     * @return resident Identity Provider
+     * @throws IdentityOAuth2Exception
+     */
+    private IdentityProvider getIdentityProvider(String tenantDomain, String jwtIssuer) throws IdentityOAuth2Exception {
+        String issuer = StringUtils.EMPTY;
+        IdentityProvider residentIdentityProvider = null;
+        try {
+            residentIdentityProvider = IdentityProviderManager.getInstance().getResidentIdP(tenantDomain);
+        } catch (IdentityProviderManagementException e) {
+            String errorMsg = String.format(ERROR_GET_RESIDENT_IDP, tenantDomain);
+            throw new IdentityOAuth2Exception(errorMsg, e);
+        }
+        FederatedAuthenticatorConfig[] fedAuthnConfigs = residentIdentityProvider.getFederatedAuthenticatorConfigs();
+        FederatedAuthenticatorConfig oauthAuthenticatorConfig =
+                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
+                        IdentityApplicationConstants.Authenticator.OIDC.NAME);
+        if (oauthAuthenticatorConfig != null) {
+            issuer = IdentityApplicationManagementUtil.getProperty(oauthAuthenticatorConfig.getProperties(),
+                    OPENID_IDP_ENTITY_ID).getValue();
+        }
+        if (jwtIssuer.equals(issuer)) {
+            return residentIdentityProvider;
+        } else {
+            return null;
+        }
```

Comment:
+1 not using Java 8 Optional in here. However, what is the possibility of using [null object pattern](https://www.tutorialspoint.com/design_pattern/null_object_pattern.htm)? However, we can proceed with null since this would be a big change. WDYT?